In [ ]:
import os
import glob
import shutil
from tqdm.notebook import tqdm

## 문장 개수

In [ ]:
total = glob.glob('/content/drive/MyDrive/anichat/tts/vits/anichat_project_removed/*.wav')
print("전체 문장 개수 = " + format(len(total), ',') + "개")

전체 문장 개수 = 3,869개


전체 문장 개수 = 3,869개

In [ ]:
con = []
for i in tqdm(total):
    if i.split('/')[8][:11] == 'anichat_con':
        con.append(i)
print("코난 파일 문장 개수 = " + format(len(con), ',') + "개")

  0%|          | 0/3869 [00:00<?, ?it/s]

코난 파일 문장 개수 = 2,710개


코난 파일 문장 개수 = 2,710개

In [ ]:
nam = []
for i in tqdm(total):
    if i.split('/')[8][:11] == 'anichat_nam':
        nam.append(i)
print("쿠도 신이치 파일 문장 개수 = " + format(len(nam), ',') + "개")

  0%|          | 0/3869 [00:00<?, ?it/s]

쿠도 신이치 파일 문장 개수 = 512개


쿠도 신이치 파일 문장 개수 = 512개

In [ ]:
you = []
for i in tqdm(total):
    if i.split('/')[8][:11] == 'anichat_you':
        you.append(i)
print("모리 코고로 파일 문장 개수 = " + format(len(you), ',') + "개")

  0%|          | 0/3869 [00:00<?, ?it/s]

모리 코고로 파일 문장 개수 = 647개


모리 코고로 파일 문장 개수 = 647개

## 단어 개수

In [ ]:
import json

In [ ]:
total_word = glob.glob('/content/drive/MyDrive/anichat/tts/vits/anichat_project_removed/*.json')
word = []
for i in tqdm(glob.glob('/content/drive/MyDrive/anichat/tts/vits/anichat_project_removed/*.json')):
    with open(i, 'r') as f:
        json_data = json.load(f)
        for j in json_data['text'].split():
            word.append(j)
print("총 단어 개수 = "+format(len(word), ',')+"개")
print("파일 당 평균 단어 개수 = {:.2f}개".format(len(word)/len(total)))

  0%|          | 0/3869 [00:00<?, ?it/s]

총 단어 개수 = 22,284개
파일 당 평균 단어 개수 = 5.76개


총 단어 개수 = 22,284개

파일 당 평균 단어 개수 = 약 5.76개

In [ ]:
set_word = set(word)
print("중복 제거한 후 단어 개수 = "+format(len(set_word), ',')+"개")

중복 제거한 후 단어 개수 = 10,416개


중복 제거한 후 단어 개수 = 10.416개

In [ ]:
con_word = []
for i in tqdm(total_word):
    if i.split('/')[8][:11] == 'anichat_con':
        with open(i, 'r') as f:
            json_data = json.load(f)
            for j in json_data['text'].split():
                con_word.append(j)
print("코난 단어 개수 = "+format(len(con_word), ',')+"개")
print("파일 당 평균 단어 개수 = {:.2f}개".format(len(con_word)/len(con)))

  0%|          | 0/3869 [00:00<?, ?it/s]

코난 단어 개수 = 15,359개
파일 당 평균 단어 개수 = 5.67개


코난 단어 개수 = 15,359개

파일 당 평균 단어 개수 = 약 5.67개

In [ ]:
set_con_word = set(con_word)
print("중복 제거한 후 단어 개수 = "+format(len(set_con_word), ',')+"개")

중복 제거한 후 단어 개수 = 7,786개


중복 제거한 후 단어 수 = 7,786개

In [ ]:
nam_word = []
for i in tqdm(total_word):
    if i.split('/')[8][:11] == 'anichat_nam':
        with open(i, 'r') as f:
            json_data = json.load(f)
            for j in json_data['text'].split():
                nam_word.append(j)
print("쿠도 신이치 단어 개수 = "+format(len(nam_word), ',')+"개")
print("파일 당 평균 단어 개수 = {:.2f}개".format(len(nam_word)/len(nam)))

  0%|          | 0/3869 [00:00<?, ?it/s]

쿠도 신이치 단어 개수 = 3,062개
파일 당 평균 단어 개수 = 5.98개


쿠도 신이치 단어 개수 = 3,062개

파일 당 평균 단어 개수 = 약 5.98개

In [ ]:
set_nam_word = set(nam_word)
print("중복 제거한 후 단어 개수 = "+format(len(set_nam_word), ',')+"개")

중복 제거한 후 단어 개수 = 1,954개


중복 제거한 후 단어 수 = 1,954개

In [ ]:
you_word = []
for i in tqdm(total_word):
    if i.split('/')[8][:11] == 'anichat_you':
        with open(i, 'r') as f:
            json_data = json.load(f)
            for j in json_data['text'].split():
                you_word.append(j)
print("모리 코고로 단어 개수 = "+format(len(you_word), ',')+"개")
print("파일 당 평균 단어 개수 = {:.2f}개".format(len(you_word)/len(you)))

  0%|          | 0/3869 [00:00<?, ?it/s]

모리 코고로 단어 개수 = 3,863개
파일 당 평균 단어 개수 = 5.97개


모리 코고로 단어 개수 = 3,863개

파일 당 평균 단어 개수 = 약 5.97개

In [ ]:
set_you_word = set(you_word)
print("중복 제거한 후 단어 개수 = "+format(len(set_you_word), ',')+"개")

중복 제거한 후 단어 개수 = 2,526개


중복 제거한 후 단어 수 = 2,526개

## 글자 개수

In [ ]:
cha = ''
for i in tqdm(total_word):
    with open(i, 'r') as f:
        json_data = json.load(f)
        cha += json_data['text'].strip()
print("총 글자 개수 = "+format(len(cha), ',')+"개")

  0%|          | 0/3869 [00:00<?, ?it/s]

총 글자 개수 = 82,916개


총 글자 개수 = 82,916개

In [ ]:
con_cha = ''
for i in tqdm(total_word):
    if i.split('/')[8][:11] == 'anichat_con':
        with open(i, 'r') as f:
            json_data = json.load(f)
            con_cha += json_data['text'].strip()
print("코난 글자 개수 = "+format(len(con_cha), ',')+"개")

  0%|          | 0/3869 [00:00<?, ?it/s]

코난 글자 개수 = 57,619개


코난 글자 개수 = 57,619개

In [ ]:
nam_cha = ''
for i in tqdm(total_word):
    if i.split('/')[8][:11] == 'anichat_nam':
        with open(i, 'r') as f:
            json_data = json.load(f)
            nam_cha += json_data['text'].strip()
print("쿠도 신이치 글자 개수 = "+format(len(nam_cha), ',')+"개")

  0%|          | 0/3869 [00:00<?, ?it/s]

쿠도 신이치 글자 개수 = 11,222개


쿠도 신이치 글자 개수 = 11,222개

In [ ]:
you_cha = ''
for i in tqdm(total_word):
    if i.split('/')[8][:11] == 'anichat_you':
        with open(i, 'r') as f:
            json_data = json.load(f)
            you_cha += json_data['text'].strip()
print("모리 코고로 글자 개수 = "+format(len(you_cha), ',')+"개")

  0%|          | 0/3869 [00:00<?, ?it/s]

모리 코고로 글자 개수 = 14,075개


모리 코고로 글자 개수 = 14,075개

## 전체 문장 길이 구하기

In [ ]:
import wave
import contextlib

In [ ]:
acc = 0
for i in tqdm(total):

 with contextlib.closing(wave.open(i,'r')) as f:
  frames = f.getnframes()
  rate = f.getframerate()
  duration = frames / float(rate)
  #print(duration)
  acc += duration #누적값

#결과 출력
print("전체 문장 길이 = 약 {:.1f}분({}초)".format(acc/60, acc))
print("파일 당 평균 문장 길이 = 약 {:.3f}초".format(acc/len(total)))

  0%|          | 0/3869 [00:00<?, ?it/s]

전체 문장 길이 = 약 196.2분(11770.064489795948초)
파일 당 평균 문장 길이 = 약 3.042초


전체 문장 길이 = 약 196.2분(11770.064489795948초)

파일 당 평균 문장 길이 = 약 3.042초

In [ ]:
con_acc = 0
for i in tqdm(total):
    if i.split('/')[8][:11] == 'anichat_con':

        with contextlib.closing(wave.open(i,'r')) as f:
            frames = f.getnframes()
            rate = f.getframerate()
            duration = frames / float(rate)
            #print(duration)
            con_acc += duration #누적값

#결과 출력
print("코난 문장 길이 = 약 {:.1f}분({}초)".format(con_acc/60, con_acc))
print("코난 파일 당 평균 문장 길이 = 약 {:.3f}초".format(con_acc/len(con)))

  0%|          | 0/3869 [00:00<?, ?it/s]

코난 문장 길이 = 약 134.2분(8053.1788208616745초)
코난 파일 당 평균 문장 길이 = 약 2.972초


코난 문장 길이 = 약 134.2분(8053.1788208616745초)

코난 파일 당 평균 문장 길이 = 약 2.972초

In [ ]:
nam_acc = 0
for i in tqdm(total):
    if i.split('/')[8][:11] == 'anichat_nam':

        with contextlib.closing(wave.open(i,'r')) as f:
            frames = f.getnframes()
            rate = f.getframerate()
            duration = frames / float(rate)
            #print(duration)
            nam_acc += duration #누적값

#결과 출력
print("쿠도 신이치 문장 길이 = 약 {:.1f}분({}초)".format(nam_acc/60, nam_acc))
print("쿠도 신이치 파일 당 평균 문장 길이 = 약 {:.3f}초".format(nam_acc/len(nam)))

  0%|          | 0/3869 [00:00<?, ?it/s]

쿠도 신이치 문장 길이 = 약 26.9분(1612.2468480725618초)
쿠도 신이치 파일 당 평균 문장 길이 = 약 3.149초


쿠도 신이치 문장 길이 = 약 26.9분(1612.2468480725618초)

쿠도 신이치 파일 당 평균 문장 길이 = 약 3.149초

In [ ]:
you_acc = 0
for i in tqdm(total):
    if i.split('/')[8][:11] == 'anichat_you':

        with contextlib.closing(wave.open(i,'r')) as f:
            frames = f.getnframes()
            rate = f.getframerate()
            duration = frames / float(rate)
            #print(duration)
            you_acc += duration #누적값

#결과 출력
print("모리 코고로 문장 길이 = 약 {:.1f}분({}초)".format(you_acc/60, you_acc))
print("모리 코고로 파일 당 평균 문장 길이 = 약 {:.3f}초".format(you_acc/len(you)))

  0%|          | 0/3869 [00:00<?, ?it/s]

모리 코고로 문장 길이 = 약 35.1분(2104.638820861681초)
모리 코고로 파일 당 평균 문장 길이 = 약 3.253초


모리 코고로 문장 길이 = 약 35.1분(2104.638820861681초)

모리 코고로 파일 당 평균 문장 길이 = 약 3.253초

## 문장별 길이

In [ ]:
dur = []
for i in tqdm(total):
    with contextlib.closing(wave.open(i,'r')) as f:
        frames = f.getnframes()
        rate = f.getframerate()
        duration = frames / float(rate)
        dur.append(duration)
dur = sorted(dur)
print("가장 짧은 문장 길이 = {:.2f}초".format(dur[0]))
print("가장 긴 문장 길이 = {:.2f}초".format(dur[-1]))

  0%|          | 0/3869 [00:00<?, ?it/s]

가장 짧은 문장 길이 = 0.23초
가장 긴 문장 길이 = 14.29초


가장 짧은 문장 길이 = 0.23초

가장 긴 문장 길이 = 14.29초

In [ ]:
con_dur = []
for i in tqdm(total):
    if i.split('/')[8][:11] == 'anichat_con':

        with contextlib.closing(wave.open(i,'r')) as f:
            frames = f.getnframes()
            rate = f.getframerate()
            duration = frames / float(rate)
            con_dur.append(duration)
con_dur = sorted(con_dur)
print("가장 짧은 코난 문장 길이 = {:.2f}초".format(con_dur[0]))
print("가장 긴 코난 문장 길이 = {:.2f}초".format(con_dur[-1]))

  0%|          | 0/3869 [00:00<?, ?it/s]

가장 짧은 코난 문장 길이 = 0.23초
가장 긴 코난 문장 길이 = 14.29초


가장 짧은 코난 문장 길이 = 0.23초

가장 긴 코난 문장 길이 = 14.29초

In [ ]:
nam_dur = []
for i in tqdm(total):
    if i.split('/')[8][:11] == 'anichat_nam':

        with contextlib.closing(wave.open(i,'r')) as f:
            frames = f.getnframes()
            rate = f.getframerate()
            duration = frames / float(rate)
            nam_dur.append(duration)
nam_dur = sorted(nam_dur)
print("가장 짧은 쿠도 신이치 문장 길이 = {:.2f}초".format(nam_dur[0]))
print("가장 긴 쿠도 신이치 문장 길이 = {:.2f}초".format(nam_dur[-1]))

  0%|          | 0/3869 [00:00<?, ?it/s]

가장 짧은 쿠도 신이치 문장 길이 = 0.30초
가장 긴 쿠도 신이치 문장 길이 = 12.00초


가장 짧은 쿠도 신이치 문장 길이 = 0.30초

가장 긴 쿠도 신이치 문장 길이 = 12.00초

In [ ]:
you_dur = []
for i in tqdm(total):
    if i.split('/')[8][:11] == 'anichat_you':

        with contextlib.closing(wave.open(i,'r')) as f:
            frames = f.getnframes()
            rate = f.getframerate()
            duration = frames / float(rate)
            you_dur.append(duration)
you_dur = sorted(you_dur)
print("가장 짧은 모리 코고로 문장 길이 = {:.2f}초".format(you_dur[0]))
print("가장 긴 모리 코고로 문장 길이 = {:.2f}초".format(you_dur[-1]))

  0%|          | 0/3869 [00:00<?, ?it/s]

가장 짧은 모리 코고로 문장 길이 = 0.42초
가장 긴 모리 코고로 문장 길이 = 12.06초


가장 짧은 모리 코고로 문장 길이 = 0.42초

가장 긴 모리 코고로 문장 길이 = 12.06초